### ISSUE TITLE
finally() is called too soon on unresolved promises

### ISSUE THREAD

**Issue by blalasaadri** [fluffynuts/synchronous-promise#15]
For `SynchronousPromise`s which aren't immediately resolved, any calls of `finally` is called before the resolution. I have built a set of tests [here](https://github.com/blalasaadri/finally-in-synchronous-promise) (including a [CI build](https://travis-ci.com/blalasaadri/finally-in-synchronous-promise)) to demonstrate the problems; they occur both with `SynchronousPromise.unresolved()` and with `.pause()`.

A small extract from those tests:
```js
const events = []

SynchronousPromise.unresolved()
  .then((result) => { events.push(`result: ${result}`) })
  .then(() => { events.push('then') })
  .finally(() => { events.push('finally') })

// events is ["finally", "result: undefined", "then"]
```
```js
const events = []

SynchronousPromise.unresolved()
  .then((result) => { events.push(`result: ${result}`) })
  .catch(() => { events.push('catch') })
  .finally(() => { events.push('finally') })

// events is ["finally", "result: undefined"]
```
```js
const events = []

const promise = SynchronousPromise.resolve('init')
  .then((result) => { events.push(`result: ${result}`) })
  .pause()
  .then(() => { events.push('resumed') })
  .finally(() => { events.push('finally') })
promise.resume()

// events is ["result: init", "finally", "resumed"]
```

Of course, using a final `.then()` rather than `.finally()` is a workaround in many cases.

Affected versions: 2.0.8 and 2.0.9

**Comment by fluffynuts** [comment]
Thanks for reporting. I'll look into it. Thanks for providing example code too.

**Comment by fluffynuts** [comment]
Fixed in df4866c6861f96c0235c3f86b9beb2696db961fc, released as 2.0.10

I've included your tests (modified only to fit mocha and my lint rules) as well as attributed the issue to you. Thanks again for reporting and providing comprehensive examples and test code -- much appreciated.

**Comment by blalasaadri** [comment]
Nice, thank you @fluffynuts!
